PelotonEppSdk/Models/EventRequest.cs

```diff
@@ -2,75 +2,100 @@
 using System.Collections.Generic;
 using System.ComponentModel.DataAnnotations;
 using System.Diagnostics.CodeAnalysis;
+using System.Linq;
+using System.Net;
 using System.Threading.Tasks;
+using System.Web;
 using PelotonEppSdk.Classes;
 using PelotonEppSdk.Enums;
 
 namespace PelotonEppSdk.Models
 {
     public class EventRequest : RequestBase
     {
-        [Required]
         [StringLength(32, MinimumLength = 32, ErrorMessage = nameof(EventToken) + " must be 32 characters long.")]
         public string EventToken { get; set; }
 
+        /// <summary>
+        /// ClientToken for the target peloton account
+        /// </summary>
+        //[Required]
```

Comment:
In short: no.

The current validation framework uses the the data annotations, but the Request objects are used for >1 request methods (GET, POST, PUT, etc).

So, for one request method the type will be required, but for another it will not be required. The current validation framework is not sufficient to have [Required] apply for specific methods.

I have work that addresses this issue in another branch, but it is still a bit of a work-in-progress. See branch validation-rework